### PR TITLE
[KOGITO-2785] - Operator nightly pipeline: add olm tests

### PR DESCRIPTION
**Pull request by Kaitou786**
See: https://issues.redhat.com/browse/KOGITO-2785
Tested to run successfully on: https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/custom/job/kaitou786/job/olm-test/9/console
Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -155,6 +159,11 @@ pipeline {
                 }
             }
         }
+        stage('Run OLM tests') {
+            steps {
+                sh 'make olm-tests'
```
Comment (radtriste) [inline Jenkinsfile.deploy:164]: This need KIND or will it be done directly into a container ?

**Review by radtriste — APPROVED** [review]
lgtm

one question though to be sure

File: Jenkinsfile.deploy (hunk 1 above)
Comment (Kaitou786) [reply Jenkinsfile.deploy:164]: The operator-test script by OLM guys install kind on there own, even if we already have `kind` on system they install there preferred version and use that :)

Comment (radtriste) [reply Jenkinsfile.deploy:164]: ok, can we comment it in the `run-olm-tests.sh` ?

Comment (Kaitou786) [reply Jenkinsfile.deploy:164]: Added a brief description :)

**Review by radtriste — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**